### PR TITLE
DOCSP-45454-oidc-ldap-shell

### PR DESCRIPTION
**Pull request by jason-price-mongodb**
## DESCRIPTION

Updates for security options per JIRA.

## STAGING

https://deploy-preview-369--docs-mongodb-shell.netlify.app/reference/options/#std-option-mongosh.--authenticationMechanism

## JIRA

https://jira.mongodb.org/browse/DOCSP-45454

## Self-Review Checklist

- [ ] Is this free of any warnings or errors in the RST?
- [ ] Is this free of spelling errors?
- [ ] Is this free of grammatical errors?
- [ ] Is this free of staging / rendering issues?
- [ ] Are all the links working?

## External Review Requirements

[What's expected of an external reviewer?](https://wiki.corp.mongodb.com/display/DE/Reviewing+Guidelines+for+the+MongoDB+Server+Documentation)

### CODE REVIEW

File: source/reference/options.txt [hunk 1]
```diff
@@ -452,15 +452,28 @@ Authentication Options
...
 
         - External authentication using OpenID Connect. This mechanism is
-          available only in `MongoDB Enterprise
-          <http://www.mongodb.com/products/mongodb-enterprise-advanced?jmp=docs>`_.
+          available in `MongoDB Enterprise
+          <http://www.mongodb.com/products/mongodb-enterprise-advanced?jmp=docs>`_
+          and `MongoDB Atlas
+          <https://www.mongodb.com/atlas/database?tck=docs_server>`_.
+
+      * - ``MONGODB-AWS`` (AWS IAM)
```
Comment (sarah-olson-mongodb) [inline source/reference/options.txt:469]: (linking)
Can you work with Fuat, Anna, or Gaurab to identify the best place to link for MONGODB-AWS information. Here is the location in Atlas, for example: https://www.mongodb.com/docs/atlas/security/aws-iam-authentication/

Also, is this needed somewhere in the server or shell docs? A quick search didn't turn up an obvious link here.

**Review by sarah-olson-mongodb — APPROVED** [review]
Thanks @jason-price-mongodb! LGTM with a question to resolve.

File: source/reference/options.txt (hunk 1 above)
Comment (jason-price-mongodb) [reply source/reference/options.txt:469]: Per our Slack convo, will ask Anna to review the PR and see if and how AWS IAM and EA need to be doc'ed.

**Comment by jason-price-mongodb** [comment]
@addaleax please review. There is the question about the AWS IAM and EA scenario and if it is required to be doc'ed and where (Sarah also added that question in the review comments above). Might be a new project external to this shell JIRA. Would like to know what you think also. Thanks, Jason.

File: source/reference/options.txt (hunk 1 above)
Comment (addaleax) [reply source/reference/options.txt:469]: I don't think I understand the question here. This PR should cover what is required for mongosh – I'd expect the server to also document this feature, similar to how Atlas does that.

**Review by addaleax — APPROVED** [review]

**Comment by jason-price-mongodb** [comment]
@GaurabAryal please review. Thanks.

**Comment by jason-price-mongodb** [comment]
Per a conversation with @sarah-olson-mongodb , Sarah will research the existing tickets and see if there is an engineering and/or docs ticket to cover the AWS and EA scenario. For now, will merge and backport the PR for this shell update. Cheers.

**Pull request merged**